pages/scan.tsx

```diff
@@ -1,111 +1,110 @@
 import React, { useState, useCallback } from "react";
 import styled from "styled-components";
 
+import { useToasts } from "react-toast-notifications";
+
 import { handleLoginRedirect, getProfile } from "../lib/authenticate";
 
 import Head from "../components/Head";
-import Navbar from "../components/Navbar";
 import Scanner from "../components/Scanner";
 
-import { Background, Container, Form, Flex } from "../styles";
+import { Form } from "../styles";
 import Select from "../components/Select";
 
 const Scan = ({ profile }) => {
   const [action, setAction] = useState(null);
   const [scannedCodes, setScannedCodes] = useState([]);
-  const [successfulScan, setSuccessfulScan] = useState(null);
+
+  // TOASTS
+  const { addToast } = useToasts();
 
   const handleActionChange = e => {
     setAction(e.target.value);
   };
 
+  const checkIfUserId = (code: string): boolean => {
+    const GOOGLE_AUTH_PREFIX = "google-oauth2|";
+    const AUTH0_PREFIX = "auth0|";
+
+    return code.startsWith(GOOGLE_AUTH_PREFIX) || code.startsWith(AUTH0_PREFIX);
+  };
+
   const sendScanRequest = async (code: string) => {
-    const idRequest = await fetch(`/api/live/identity-check/${code}`);
-
-    if (idRequest.status === 200) {
-      // We successfully ID'd a user, show the first and last name to the scanner
-      const idData = await idRequest.json();
-
-      // Send a request to the server to scan hacker for task
-      const scanRequest = await fetch("/api/live/dispatch", {
-        method: "POST",
-        body: JSON.stringify({
-          userId: code,
-          actionId: "checkin"
-        }),
-        headers: {
-          "Content-Type": "application/json"
-        }
-      });
+    if (!checkIfUserId(code)) {
+      return;
+    }
 
-      if (scanRequest.status === 200) {
-        // Successful scan, let's display that to the user
-        setSuccessfulScan({
-          firstName: idData.firstName,
-          lastName: idData.lastName
-        });
+    // Send a request to the server to scan hacker for task
+    const scanRequest = await fetch("/api/live/dispatch", {
+      method: "POST",
+      body: JSON.stringify({
+        userId: code,
+        actionId: action === "checkin" ? "checkin" : "contrib",
```

Comment:
Another enum chance, especially since it's TS. Can just to odyssey.d.ts

Reply:
yeah, i'm probably going to address some of this in a later PR -- mostly wrote this to do a quick test for doing QR scans for both checkin and contributions